File: backend/express/src/models/user.model.js
import mongoose from "mongoose";

const UserSchema = new mongoose.Schema({
  name: { type: String, required: true },
  phone: { type: String, unique: true },
  email: { type: String, unique: true },
  age: { type: Number, required: true },
  gender: { type: String, required: true },
  height: { type: String, required: true },
  weight: { type: String, required: true },
  dateOfBirth: { type: Date, required: true },
  maritalStatus: { type: String, required: true },
  bloodGroup: { type: String, required: true },
  allergies: { type: String, required: true },
  medicalConditions: { type: String, required: true },
  medications: { type: String, required: true },
  address: { type: String, required: true },
  city: { type: String, required: true },
  state: { type: String, required: true },
  country: { type: String, required: true },
  zipCode: { type: String, required: true },

  profilePic: String,
  
  emergencyContact: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "Contact" // Assigned ONE contact for SOS
  },

  contacts: [{ 
    type: mongoose.Schema.Types.ObjectId, 
    ref: "Contact" 
  }]
});

module.exports = mongoose.model("User", UserSchema);

File: backend/spring/demo/src/main/java/com/springbootbackend/demo/model/UserModel.java
package com.springbootbackend.demo.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Date;
import java.util.List;
@NoArgsConstructor
@AllArgsConstructor
@Data
@Document(collection = "users")
public class UserModel {
    @Id
    private String id;

    private String name;
    private int age;
    private String email;
    private String phoneNo;
    private String gender;
    private String height;
    private String weight;
    private Date dateOfBirth;
    private String maritalStatus;
    private String bloodGroup;
    private String allergies;
    private String medicalConditions;
    private String medications;
    private String address;
    private String city;
    private String state;
    private String country;
    private String zipCode;

    private String profilePic;

    @DBRef
    private Contact emergencyContact;

    @DBRef
    private List<Contact> contacts;
    @DBRef
    private List<MedicalRecord> medicalRecords;
    @DBRef
    private List<EmergencyAlert> emergencyAlerts;
    @DBRef
    private List<VideoCallLog> videoCallLog;

}